### PR TITLE
dbt-materialize: continous testing using --store-failures

### PR DESCRIPTION
**Pull request by ahelium**
### Motivation

  * This PR adds a feature that has not yet been specified.
   
   This work came out a conversation with [jwills](https://github.com/jwills) in the community [slack](https://materializecommunity.slack.com/archives/C038XJM0GP7/p1651509389762489): 

> Pitch: With MZ, I can execute dbt run once-- not repeatedly-- and my data will continually stay up to date whenever I query it. I want to do the same thing for dbt test-- a way where I can run a single command and be immediately notified in whatever modality I want (Slack, email, etc.) when the data changes in such a way that one of my tests fails

  dbt provides a `--store-failures flag`. In batch dwh environments, this (at time of test) creates a table per test condition. Materialize should, instead, create materialized views. 

### Tips for reviewer

This is a WIP because I'd like to solicit feedback from everyone who has touched this adapter! 

If we implement this `materializedtest` custom materialization, we will be diverging even further from the underlying dbt-postgres package. I'm proposing that, in this case, it is appropriate to do so. We _should_ enable a different type of test and package it as such. I'm very open to different interpretations. 

Other notes:
`dbt-test (without configuring a materializedtest)` works as intended, but `dbt-test (without configuring a materializedtest) --store-failures` does not. Out of the box support would be blocked on https://github.com/MaterializeInc/materialize/issues/5266. Seems important to document or handle for folks.

If we end up merging something, we should bump the adapter version and update all of the correct docs. 

### Testing

I built a docker image locally and mocked up how I'd use this in our existing dbt-getting-started [demo](https://github.com/MaterializeInc/demos/pull/31).

### CODE REVIEW

**Comment by benesch** [comment]
Huge +1 on the idea, but can we just override the built-in `test` materialization, rather than making a new `materializedtest` one? Then users would get the magic auto-updating tests without any extra work. I can't think of any downsides.

**Comment by morsapaes** [comment]
It's pretty neat that you found a way to do this without depending on any external services! 🎉  Need some time to digest the whole thing, also to understand where we should draw the line IWRT **scope** for continuous testing. Using it for generic alerting (like in that demo example) seems to...be mixing things up a bit. What concrete value should users expect from this? How do we intend them to use (and hopefully not abuse) this?

At first glance, I agree with Nikhil that we should try to override and/or build on the default behaviour of `dbt test` instead of spinning up a separate implementation. It could also be a good idea to bring it up for discussion in the dbt community (in #adapter-ecosystem) to get some validation and feedback before committing to an implementation.

**Comment by ahelium** [comment]
Right on! Thank you feedback - 100% agree about overriding `dbt test`. I'm inclined to just go for releasing this, since the store_failures flag was unusable out of the box anyway and we've made clear the limitations the dbt-materialize adapter has wrt `tables`.

@morsapaes you make sense. I'll keep the demo update as a WIP and let's discuss there.

**Comment by morsapaes** [comment]
SGTM, @ahelium! And sorry, I didn't get that `--store-failures` was an unused flag; didn't mean to block but totally sounded like I did. Can you take care of adding the docs on the dbt side? There's an [open PR](https://github.com/dbt-labs/docs.getdbt.com/pull/1401) to add a Materialize configs page, it'd fit well in there.

File: misc/dbt-materialize/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # dbt-materialize Changelog
 
+## 1.1.1 - 2022-05-04
+
+* Provide support for storing the results of a test query in a `materializedview` using `store_failures`.
```
Comment (morsapaes) [inline misc/dbt-materialize/CHANGELOG.md:5]:
```suggestion
* Provide support for storing the results of a test query in a `materializedview` using the [`store_failures` config](https://docs.getdbt.com/reference/resource-configs/store_failures).
```

**Review by morsapaes — APPROVED** [review]

File: misc/dbt-materialize/dbt/adapters/materialize/__version__.py [hunk 2]
```diff
@@ -15,4 +15,4 @@
 # limitations under the License.
 
 # If you bump this version, bump it in setup.py too.
-version = "1.1.0"
+version = "1.1.1"
```
Comment (morsapaes) [inline misc/dbt-materialize/dbt/adapters/materialize/__version__.py:18]: You need to bump it in `setup.py` as well.

**Pull request merged**